pkg/config/apiserver.go

```diff
@@ -278,14 +322,55 @@ func (a *API) putRules(req *restful.Request, resp *restful.Response) {
 	}
 
 	if _, err = a.store.Set(key, val); err != nil {
+		glog.Warningf("Failed to save %s\n %s", key, cerr.Error())
 		writeErrorResponse(http.StatusInternalServerError, err.Error(), resp)
 		return
 	}
 	// TODO send index back to the client
-	writeResponse(http.StatusOK, fmt.Sprintf("Created %s", key),
-		vd.rule[*parseRulesKey(key)], resp)
+	writeResponse(http.StatusOK, fmt.Sprintf("Created %s", key), getreply(vd), resp)
+}
+
+// putRules replaces the entire rules document for the scope and subject
+// "/scopes/{scope}/subjects/{subject}/rules"
+func (a *API) putRules(req *restful.Request, resp *restful.Response) {
+	a.storeConfigIfValid(req, resp, func(vd *Validated) interface{} {
+		return vd.rule[rulesKey{
+			Scope:   req.PathParameter("scope"),
+			Subject: req.PathParameter("subject")}]
+	})
+}
+
+// putAdaptersOrDescriptors creates or replaces adapter configurations.
```

Comment:
nit: maybe just drop "adapter" from all of these comments and just say "configurations."